impl/src/main/java/org/jboss/cdi/tck/tests/beanContainer/assignability/BeanEventAssignabilityTest.java

```diff
@@ -0,0 +1,186 @@
+        assertThrows("Null required type", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingBean(beanTypes, Set.of(), null, Set.of()));
+        assertThrows("Null required qualifiers", IllegalArgumentException.class, () -> getCurrentBeanContainer().isMatchingBean(beanTypes, Set.of(), MyBean.class, null));
+    }
+
+    @Test
+    @SpecAssertion(section = BM_BEAN_EVENT_ASSIGNABILITY, id = "ad")
+    public void testEventMatching() {
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEvent.class, Set.of()),
+                "Event did not match its own type");
+
+        assertFalse(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE)),
+                "Event matched despite not having required qualifier");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE), MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE)),
+                "Event did not match despite having required qualifier");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE, Qualifier2.Literal.INSTANCE), MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE)),
+                "Event did not match despite having a superset of the required qualifiers");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEventInterface.class, Set.of()),
+                "Event should match when a supertype is required");
+        
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), Object.class, Set.of()),
+                "Event should match when Object is required");
+        
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Default.Literal.INSTANCE), MyEvent.class, Set.of()),
+                "Event with @Default should match when no qualifiers required");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Any.Literal.INSTANCE), MyEvent.class, Set.of()),
+                "Event with explicit @Any should match when no qualifiers required");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(Qualifier1.Literal.INSTANCE), MyEvent.class, Set.of()),
+                "Event with @Qualifier1 should match when no qualifiers required");
+
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(NamedLiteral.of("foo")), MyEvent.class, Set.of()),
+                "Event with @Named should match when no qualifiers required");
+
+    }
+    
+    @Test
+    @SpecAssertion(section = BM_BEAN_EVENT_ASSIGNABILITY, id = "ad")
+    public void testEventMatchingDefaultQualifier() {
+        assertTrue(getCurrentBeanContainer().isMatchingEvent(MyEvent.class, Set.of(), MyEvent.class, Set.of(Default.Literal.INSTANCE)),
```

Comment:
@Azquelt I believe this is incorrect and should be `assertFalse`. With observers, you are *not* adding `@Default` if missing, just `@Any`.

Reply:
Oh, you are right, I completely forgot about that. I've tweaked Weld PR accordingly and it now passes all of these tests.

> Events don't get @Default as an extra qualifier ...

Well, the spec bit you posted basically implies they do get `@Default` when no other qualifier is present. Otherwise you wouldn't satisfy the observer resolution condition of OM having a subset of qualifiers that the event has.

Reply:
Yeah, I _think_ the two end up equivalent, which makes it all the more strange that they're specified differently.

Reply:
I agree. Either way, marking this as resolved.
Thanks again for pointing it out

Reply:
We should probably mention this in the javadoc of `isMatchingEvent()` as well. I'll add something to https://github.com/jakartaee/cdi/pull/748